snappy/snappy.go

```diff
@@ -0,0 +1,55 @@
+package snappy
+
+import (
+	"bytes"
+
+	"github.com/eapache/go-xerial-snappy"
+	kafka "github.com/segmentio/kafka-go"
+)
+
+func init() {
+	kafka.RegisterCompressionCodec(func() kafka.CompressionCodec {
+		return NewCompressionCodec()
+	})
+}
+
+type CompressionCodec struct{}
+
+func NewCompressionCodec() CompressionCodec {
+	return CompressionCodec{}
+}
+
+// Code implements the kafka.CompressionCodec interface.
+func (c CompressionCodec) Code() int8 {
+	return 2
+}
+
+// Encode implements the kafka.CompressionCodec interface.
+func (c CompressionCodec) Encode(dst, src []byte) (int, error) {
+	buf := buffer{data: dst}
+	return buf.Write(snappy.Encode(src))
```

Comment:
Honestly the framing header is so simple in https://github.com/eapache/go-xerial-snappy/blob/master/snappy.go that I feel like we should just bring the code in and not take a dependency to the go-xerial-snappy package here.

This would allow you to use a similar implementation to the one you used for gzip and lz4 based on the snappy reader and writer in https://godoc.org/github.com/golang/snappy, which would be more efficient than forcing the allocation of a temporary buffer like what's currently done.